### DISCUSSION TITLE
Lock file maintenance is executed on multiple branches

### DISCUSSION THREAD

**Discussion by ahippler** [renovatebot/renovate#12324] [Request Help]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Describe the bug

when lockfile maintenance is executed is executed multiple times when adding a package rule with a group name and matchPaths. The PR is a mix of the expected update and lockfile maintenance.

I created a repo to reproduce the bug: https://github.com/ahippler/renovate-multiple-lockFileMaintenance/pull/2

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by ahippler** [comment]
i think the bug happens in https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/updates/flatten.ts#L113. the `supportsLockFileMaintenance` is done for every `packageFileConfig` and not for every manager or lockfile.

**Comment by rarkins** [comment]
Seems like `matchPaths` is matching the lock file maintenance update?

```
    {
      "groupName": "a deps",
      "matchPaths": ["modules/*/package.json"],
      "enabled": true
    }
```

Next we need to work out if someone might be *relying* on that for legitimate reasons.

**Comment by ahippler** [comment]
its the combination of `matchPaths`and `groupName`. I dont think it is intended/legit this way because you can not control it. I did not find a way to disable it without disabling the lockfile maintenance at all.

**Comment by rarkins** [comment]
You could probably use matchUpdateTypes

**Comment by ahippler** [comment]
setting matchUpdateTypes does not help. will also only disable or enable the whole package rule. the updateType is set to lockfileMaintenance for alle updates matching the packageRule with matchPaths. thats why the updated versions in the PR (https://github.com/ahippler/renovate-multiple-lockFileMaintenance/pull/2) show `All locks refreshed` instead of the changed versions. Its really a serious bug.

**Comment by rarkins** [comment]
I don't agree that it's a serious bug. I recommend that you disable lock file maintenance if you can't find any workaround.

**Reply by rarkins** [reply]
See https://github.com/renovate-reproductions/12317/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc
